sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -484,12 +496,24 @@ case class AdaptiveSparkPlanExec(
    * Notify the listeners of the physical plan change.
    */
   private def onUpdatePlan(executionId: Long): Unit = {
+    if (isSubquery) {
+      onUpdateAccumulator(collectSQLMetrics(currentPhysicalPlan))
```

Comment:
we can pass the `executionId` parameter to the `onUpdateAccumulator` method.

Reply:
We can not put this code in `onUpdatePlan`, Because when the sql is subquery, the `executionId` is None and it will not call the `OnUpdatePlan`.

Reply:
then can we not make it None now that we have this `if` branch??

Reply:
btw, without setting executeID, this `onUpdatePlan` will not be called at all when `isSubquery == true`.

Reply:
updated.